src/plugins/profiling/server/routes/search_flamechart.ts

```diff
@@ -271,71 +271,75 @@ async function queryFlameGraph(
     logger.info('unique downsampled stacktraces: ' + stackTraceEvents.size);
   }
 
-  const resStackTraces = await logExecutionLatency(
+  const nQueries = 4;
+  const results = new Array(nQueries);
+
+  await logExecutionLatency(
     logger,
     'mget query for ' + stackTraceEvents.size + ' stacktraces',
     async () => {
-      if (testing) {
-        return await client.search(
-          {
+      const functions = new Array(nQueries);
+      const promises = new Array(nQueries);
+      const chunkSize = Math.floor(stackTraceEvents.size / nQueries);
+      const stackTraceIDs = [...stackTraceEvents.keys()];
+
+      logger.info('A');
```

Comment:
This is just for me to see some millisecond timings. Will be removed before merging.